controllers/disruption_controller.go

```diff
@@ -746,16 +748,26 @@ func (r *DisruptionReconciler) handleChaosPodTermination(instance *chaosv1beta1.
 		r.recordEventOnDisruption(instance, chaosv1beta1.EventDisruptionStuckOnRemoval, "")
 
 		instance.Status.IsStuckOnRemoval = true
+
+		r.updateTargetInjectionStatus(instance, chaosPod, chaostypes.DisruptionInjectionStatusIsStuckOnRemoval)
 	}
 }
 
+func (r *DisruptionReconciler) updateTargetInjectionStatus(instance *chaosv1beta1.Disruption, chaosPod corev1.Pod, status chaostypes.DisruptionInjectionStatus) {
+	targetInjection := instance.Status.TargetInjections[chaosPod.Labels[chaostypes.TargetLabel]]
+	targetInjection.InjectionStatus = status
+	targetInjection.Since = instance.Spec.Duration.Duration().Milliseconds()
```

Comment:
I think I don't understand this change. The `Status.Since` is supposed to contain the timestamp of when the disruption was injected into that target. You're getting the `Spec.Duration` field value which is the maximum duration allowed for the disruption after which it'll clean itself. This value does not change over time does not represent anything related to the injection status.

What I was mentioning in my PR comment was to use pod conditions instead. On [an injector pod](https://pkg.go.dev/k8s.io/api/core/v1#Pod), and more specifically [in its status](https://pkg.go.dev/k8s.io/api/core/v1#PodStatus), you can get [its pod conditions](https://pkg.go.dev/k8s.io/api/core/v1#PodCondition). By looping over those conditions, you can search for the [PodReady](https://pkg.go.dev/k8s.io/api/core/v1#PodConditionType) one and get its `LastTransitionTime` value which is when the injector pod went to `Ready` (meaning that the readiness probe was successful, meaning that the injection was successfully done).

By the way, based on the example you pasted in the PR description, it is probably better to declare this field as a [metav1.Time](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Time) (like done in the pod condition) so on describe it displays a nice datetime format and not a timestamp number which is impossible to interpret as a human.

Reply:
Sorry, I don't understand well... 🙈 It's more clear for me 👍

Reply:
Is it important to check if the pod status went ready because the pod status is defined by the function parameter? 
Is it correct if I just take the last element of the `Status.Conditions` slice and just use the value of the **LastTransitionTime** attribute?